packages/smarthr-ui/src/components/InformationPanel/InformationPanel.tsx

```diff
@@ -118,51 +128,124 @@ export const InformationPanel: FC<Props & Omit<BaseElementProps, keyof Props>> =
-  })
+  const styles = useMemo(() => {
+    const withActive = informationPanel({
+      type,
+      active: true,
+      bold,
+    })
+    const withInactive = informationPanel({
+      type,
+      active: false,
+      bold,
+    })
+
+    const wrapperProps = { className }
+
+    return {
+      active: {
+        wrapper: withActive.wrapper(wrapperProps),
+        header: withActive.header(),
+        heading: withActive.heading(),
+        togglableButton: withActive.togglableButton(),
+        content: withActive.content(),
+      },
+      inactive: {
+        wrapper: withInactive.wrapper(wrapperProps),
+        header: withInactive.header(),
+        heading: withInactive.heading(),
+        togglableButton: withInactive.togglableButton(),
+        content: withInactive.content(),
+      },
+    }
+  }, [bold, type, className])
+
+  const currentStyles = styles[active ? 'active' : 'inactive']
 
   return (
-    <Base {...props} overflow="hidden" as="section" className={wrapper({ className })}>
-      <Cluster align="center" justify="space-between" className={header()}>
+    <Base {...props} overflow="hidden" as="section" className={currentStyles.wrapper}>
+      <Cluster align="center" justify="space-between" className={currentStyles.header}>
         {/* eslint-disable-next-line smarthr/a11y-heading-in-sectioning-content */}
-        <Heading type="blockTitle" tag={titleTag} id={titleId} className={heading()}>
-          <ResponseMessage type={type} iconGap={0.5}>
-            {title}
-          </ResponseMessage>
-        </Heading>
+        <MemoizedHeading tag={titleTag} id={titleId} className={currentStyles.heading} type={type}>
+          {title}
+        </MemoizedHeading>
         {togglable && (
-          <Button
-            suffix={active ? <FaCaretUpIcon /> : <FaCaretDownIcon />}
-            size="s"
-            onClick={handleClickTrigger}
-            aria-expanded={togglable ? active : undefined}
-            aria-controls={contentId}
-            className={togglableButton()}
-          >
-            {active
-              ? decorators?.closeButtonLabel?.(CLOSE_BUTTON_LABEL) || CLOSE_BUTTON_LABEL
-              : decorators?.openButtonLabel?.(OPEN_BUTTON_LABEL) || OPEN_BUTTON_LABEL}
-          </Button>
+          <TogglableButton
+            active={active}
+            onClickTrigger={onClickTrigger}
+            setActive={setActive}
+            contentId={contentId}
+            className={currentStyles.togglableButton}
+            decorators={decorators}
+          />
         )}
       </Cluster>
-      <div id={contentId} aria-hidden={!active} className={content()}>
+      <div id={contentId} aria-hidden={!active} className={currentStyles.content}>
         {children}
       </div>
     </Base>
   )
 }
+
+const MemoizedHeading = React.memo<
+  Pick<Props, 'type'> & {
+    tag: Props['titleTag']
+    id: string
+    className: string
+    children: Props['title']
+  }
+>(({ type, children, ...rect }) => (
```

Comment:
```suggestion
>(({ type, children, ...rest }) => (
```

typoかも？ (さらにrestよりpropsのほうが好ましいかも？)

Reply:
あ、typoですね...
あとrestは↑で書いているような残りの値をまとめる際よく使われる変数名というイディオムです。
https://typescriptbook.jp/reference/functions/rest-parameters

以前smarthr-ui内でprops or restで議論が会った際、restのほうが残余引数であることを伝えられるよね (propsだとコンポーネントに渡された引数全部の場合も考慮してしまう名前だよね)という感じでrest良さそう、となった記憶があります。

Reply:
既存のものを置き換えるほどではないですが、新規の場合はrestにしようかな、くらいの温度感だったはずなのでここもrestでいかせていただければ 🙏

Reply:
なるほどです！rest vs props論争があってのrest終結なら異論ないのでrestで！